packages/components/checkbox/src/CheckboxInput.tsx

```diff
@@ -46,56 +39,52 @@ interface RadixProps {
    * The name of the checkbox. Submitted with its owning form as part of a name/value pair.
    */
   name?: string
+  /**
+   * The value given as data when submitted with a name.
+   */
+  value?: string
+  /**
+   * Event handler called when the checked state of the checkbox changes.
+   */
+  onCheckedChange?: (checked: boolean, indeterminate?: boolean) => void
 }
 
-const useIcon = ({ icon, checked }: { icon: ReactNode; checked: AriaCheckedStatus }) => {
-  if (icon) {
-    return icon
-  }
-  switch (checked) {
-    case 'true':
-      return <Check />
-    case 'mixed':
-      return <Minus />
-    default:
-      return null
-  }
-}
-
-export interface InputProps
-  extends RadixProps, // Radix props
-    InputStylesProps, // CVA props (variants)
-    Omit<ButtonHTMLAttributes<HTMLButtonElement>, 'value' | 'checked' | 'defaultChecked'> {} // Native HTML props
+export const CheckboxInput = forwardRef<HTMLButtonElement, CheckboxInputProps>(
+  (
+    {
+      className,
+      icon = <Check />,
+      inderteminateIcon = <Minus />,
+      intent,
+      checked,
+      onCheckedChange,
+      ...others
+    },
+    ref
+  ) => {
+    const handleCheckedChange = (status: CheckedStatus) => {
+      const isChecked = status === 'indeterminate' ? false : status
+      const isIndeterminate = status === 'indeterminate' ? true : undefined
 
-export const Input = forwardRef<ElementRef<typeof CheckboxPrimitive.Root>, InputProps>(
-  ({ intent, icon, className, onCheckedChange, ...props }, forwardedRef) => {
-    const [innerChecked, setInnerChecked] = useState<'true' | 'false' | 'mixed'>()
-    const ref = useMergeRefs(forwardedRef, node => {
-      setInnerChecked(node?.getAttribute('aria-checked') as AriaCheckedStatus)
-    })
-    const innerIcon = useIcon({ icon, checked: innerChecked })
-
-    const handleOnCheckedChange = useCallback((value: CheckedStatus) => {
-      if (typeof onCheckedChange === 'function') {
-        const [checked, indeterminate] = [
-          value === 'indeterminate' ? false : value,
-          value === 'indeterminate' ? true : undefined,
-        ]
-        onCheckedChange(checked, indeterminate)
+      if (onCheckedChange) {
+        onCheckedChange(isChecked, isIndeterminate)
       }
-    }, [])
+    }
 
     return (
-      <CheckboxPrimitive.Root
+      <CheckboxPrimitive
         ref={ref}
-        className={inputStyles({ intent, className })}
-        {...props}
-        onCheckedChange={handleOnCheckedChange}
+        className={checkboxInputStyles({ intent, className })}
+        checked={checked}
+        onCheckedChange={handleCheckedChange}
+        {...others}
       >
-        <CheckboxPrimitive.Indicator className="text-surface flex items-center justify-center">
-          {innerIcon}
-        </CheckboxPrimitive.Indicator>
-      </CheckboxPrimitive.Root>
+        <CheckboxIndicator>
+          {checked === 'indeterminate' ? inderteminateIcon : icon}
```

Comment:
```suggestion
          {icon}
```

in that case this will be simpler

Reply:
We agreed that indeterminateIcon would be better